docs/test_specification.yaml

```diff
@@ -0,0 +1,218 @@
+definitions: 
```

Comment:
Be sure to configure your editor to not insert whitespace at the ends of lines. There are quite a few of these in this file.